docs/utilities/tags.md

```diff
@@ -0,0 +1,53 @@
+Tags
+====
+
+Tags are generalized sets of objects in the game, used for grouping related things together and providing fast membership checks.
+
+Declaring Your Own Groupings
+----------------------------
+Tags are declared in your mod's datapack. For example, `/data/modid/tags/blocks/foo/tagname.json` will declare a `Tag<Block>` with ID `modid:foo/tagname`.
+Similarly, you may append to or override tags declared in other domains, such as vanilla, by declaring your own JSONs.
+For example, to add your own mod's saplings to the vanilla sapling tag, you would specify it in `/data/minecraft/tags/blocks/saplings.json`, and vanilla will merge everything into one Tag at reload.
+See the [vanilla wiki](https://minecraft.gamepedia.com/Tag#JSON_format) for a description of the base syntax.
+
+Forge provides two extensions on the vanilla syntax:
+* You may declare an `optional` array of the same format as the `values` array, but any values listed here that are not present will not cause the Tag loading to error.
+This is useful to provide integration for mods that may or may not be present at runtime.
+* You may declare a `remove` array of the same format as the `values` array. Any values listed here will be removed from the Tag. This acts as a finer grained version of the vanilla `replace` option.
+
+
+Using Tags In Code
+------------------
+Block, Item, and Fluid tags are automatically sent from the server to any remote clients on login and reload. Function tags are not synced.
+
+`BlockTags.getCollection()` and `ItemTags.getCollection()` will retrieve the current `TagCollection`, from which you can retrieve a Tag object by its ID.
+With a `Tag` object in hand, membership can be tested with `tag.contains(thing)`, or all the objects in the tag queried with `tag.getAllElements()`.
+
+As an example:
+```Java
+ResourceLocation myTagId = new ResourceLocation("mymod", "myitemgroup");
+Item unknownItem = stack.getItem();
+boolean isInGroup = ItemTags.getCollection().getOrCreateTag(myTagId).contains(unknownItem);
+// alternatively, can use getTag and perform a null check
+```
+
+!!! note:
+    The `TagCollection` returned by `getCollection()` (and the `Tag`s within it) may expire if a reload happens, so you should always query the collection anew every time you need it.
+    The static `Tag` fields in `BlockTags` and `ItemTags` avoid this by introducing a wrapper that handles this expiring.
+
+
+Conventions
+-----------
+
+There are several conventions that will help facilitate compatibility in the ecosystem:
+* If there is a vanilla tag that fits your block or item, add it to that tag. See the [list of vanilla tags](https://minecraft.gamepedia.com/Tag#List_of_tags).
```

Comment:
The capitalization of `tag` isn't entirely consistent across the article, unless you wanted to distinguish between the abstract concept of "tag" and the code-linked "Tag", in which case I'd say the latter should also always be in inline code markup.